FileSystemManager/Controllers/FileSystemController.cs

```diff
@@ -329,39 +435,81 @@ public IActionResult RenameItem(string path, string newName)
                 return BadRequest("Invalid path or new name");
             }
 
-            var decodedPath = System.Net.WebUtility.UrlDecode(path);
-            var fullPath = Path.Combine(rootPath, decodedPath.TrimStart('/'));
-            var newFullPath = Path.Combine(Path.GetDirectoryName(fullPath), newName);
+            var decodedPath = System.Net.WebUtility.UrlDecode(path).TrimStart('/').Replace('\\', '/');
+            var fullPath = Path.Combine(rootPath, decodedPath).Replace('\\', '/');
+            var extension = Path.GetExtension(fullPath);
 
-            try
+            if (!newName.EndsWith(extension, StringComparison.OrdinalIgnoreCase))
             {
-                if (Directory.Exists(fullPath))
-                {
-                    Directory.Move(fullPath, newFullPath);
-                }
-                else if (System.IO.File.Exists(fullPath))
+                newName = Path.ChangeExtension(newName, extension);
+            }
+
+            var newFullPath = Path.Combine(Path.GetDirectoryName(fullPath), newName).Replace('\\', '/');
+
+            using (var transaction = _context.Database.BeginTransaction())
+            {
+                try
                 {
-                    System.IO.File.Move(fullPath, newFullPath);
+                    bool itemExistsLocally = false;
```

Comment:
why use this if we can return NotFound() in else condition if not found?

Reply:
why use this if we can return NotFound() in else condition if not found? Discuss about this condition is true 
now we need to same condition on the db level if any person delete manullay what should we do In that case